core/templates/dev/head/pages/exploration_editor/EditorServices.js

```diff
@@ -89,14 +104,29 @@ oppia.factory('explorationData', [
           // changes applied. This makes a force-refresh necessary when changes
           // are discarded, otherwise the exploration-with-draft-changes
           // (which is cached here) will be reused.
+          console.log(errorCallback);
           return (
             EditableExplorationBackendApiService.fetchApplyDraftExploration(
             explorationId).then(function(response) {
               $log.info('Retrieved exploration data.');
               $log.info(response);
               draftChangeListId = response.draft_change_list_id;
               explorationData.data = response;
-
+              var draftChanges = LocalStorageService.getExplorationDraft(
```

Comment:
It's confusing to see `draftChanges.draftChanges`. Maybe just call this `draft`?

Also, consider creating a domain object for ExplorationDraft with private properties `changeList` and `id`. It has methods toJson() on its prototype, and a createFromJson() method. That way you are passing around an actual domain object representing a draft, and not just a freely-mutable dict.